File: src/main/java/ru/tykvin/hermes/file/storage/FileBinding.java
package ru.tykvin.hermes.file.storage;

import lombok.Data;
import lombok.RequiredArgsConstructor;
import ru.tykvin.hermes.file.model.DownloadingEntity;
import ru.tykvin.hermes.model.User;

import java.time.LocalDateTime;

@Data
@RequiredArgsConstructor
public class FileBinding {
    private final DownloadingEntity downloadingEntity;
    private final User user;
    private final FileBindingConstraints constraints;
    private final LocalDateTime createAt;
}

File: src/main/kotlin/ru/tykvin/hermes/storage/FileBinding.kt
package ru.tykvin.hermes.storage

import ru.tykvin.hermes.auth.controller.User
import ru.tykvin.hermes.file.controller.DownloadingEntity
import java.time.LocalDateTime

class FileBinding(
        val downloadingEntity: DownloadingEntity,
        val user: User,
        val constraints: FileBindingConstraints,
        val createAt: LocalDateTime)
